sql/src/main/java/io/crate/operation/join/NestedLoopOperation.java

```diff
@@ -512,9 +522,64 @@ public void resume(boolean async) {
         }
     }
 
+    private class RightJoinRightRowReceiver extends RightRowReceiver {
+
+        private final BitSet matchedRows = new BitSet();
+        private long rowPosition = -1;
+
+        RightJoinRightRowReceiver(int leftNumOutputs, int rightNumOutputs) {
+            super(leftNumOutputs, rightNumOutputs);
+        }
+
+        @Override
+        public Result setNextRow(final Row rightRow) {
+            if (!stop) {
```

Comment:
I want to just increase the rowPosition if !stop and avoid copying code from superclass.